tests/checkpointing/test_model_checkpoint.py

```diff
@@ -986,3 +986,43 @@ def __init__(self, hparams):
     else:
         # make sure it's not AttributeDict
         assert type(ckpt[model.CHECKPOINT_HYPER_PARAMS_KEY]) == hparams_type
+
+
+@pytest.mark.parametrize('max_epochs', [3, 4])
+@pytest.mark.parametrize(
+    'save_top_k, expected',
+    [
+        (1, ['curr_epoch.ckpt']),
+        (2, ['curr_epoch.ckpt', 'curr_epoch-v0.ckpt']),
```

Comment:
I am thinking if in such case we shall rename the previous one
```suggestion
        (2, ['curr_epoch.ckpt-v0', 'curr_epoch-v1.ckpt']),
```
cc: @PyTorchLightning/core-contributors

Reply:
I like it. Just thinking about how to do it the right way.

Reply:
I also like the idea, solves what I commented here https://github.com/PyTorchLightning/pytorch-lightning/pull/4861#issuecomment-734458571. But better to do it in another PR, right?

Reply:
basically when you find existing and your internal counter is the min (you run counter from 1 not 0) then the fist you move to v0, it shall not be so hard, just not sure if fssspec allows it do simply...

Reply:
thanks, will try to get this done today.

Reply:
is this implemented?

Reply:
ok I see #5000

Reply:
it's wip in a separate PR. https://github.com/PyTorchLightning/pytorch-lightning/issues/5000

Reply:
The PR is https://github.com/PyTorchLightning/pytorch-lightning/pull/5008 (not WIP)